components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -20,6 +20,7 @@ interface BraveWalletProvider {
   AddEthereumChain(string json_payload) => (bool success, int32 error, string message);
   AddUnapprovedTransaction(TxData tx_data, string from) => (bool success, string tx_meta_id, string error_message);
   AddUnapproved1559Transaction(TxData1559 tx_data, string from) => (bool success, string tx_meta_id, string error_message);
+  SignMessage(string address, string message) => (bool success, string signature, int32 error, string error_message);
```

Comment:
maybe optional error params and optional signature? Could probably get rid of success if so too, but ok to keep too.  If you prefer to leave it as is I'm fine with that too but it's a bit strange to have both int32 error and a bool success.

Reply:
signature is the main result, it can't be optional.
removing `bool success` in 4948f7ce356351ef501571c305ebb9fa57e5c719 and 804931bf939decfd96274cbb1327362f326e2b41

Reply:
I was thinking if it fails it could be no result so that's why optional. But it's ok like this.